tmol/score/elec/params.py

```diff
@@ -38,20 +40,32 @@ class ElecParamResolver(ValidateAttrs):
     # map (AA,atom) to atom
     cp_reps: dict
 
-    cp_reps_by_res: dict
-
     # map (AA,atom) to partial charge
     partial_charges: dict
 
+    # fd this can go away when old scoresystem is removed
     def resolve_partial_charge(
         self, res_names: NDArray[object][:, :], atom_names: NDArray[object][:, :]
     ) -> NDArray[numpy.float32][...]:
         """Convert array of atom type names to partial charges."""
-        pcs = numpy.vectorize(
-            lambda a, b: self.partial_charges[(a, b)], otypes=[numpy.float32]
-        )(res_names, atom_names)
+
+        def lookup_charge(res, atm):
+            if res is None:
+                return 0.0
+            tag, *vars = res.split(":")
+            vars.append("")
+            for vi in vars:
```

Comment:
Good point.  This does create an order dependance....

I can change this code so that it ensures that no two _applied_ patches have different charges for the same atoms?